### ISSUE TITLE
feat: train MetricAdapter with Triplet Loss — G11 training loop

### ISSUE THREAD

**Issue by samscarrow** [git-scarrow/fast-associative-memory#12]
## Objective
Train the `MetricAdapter` (implemented in PR #9) using Online Hard Negative Mining so that fine-grained sibling classes (e.g. Husky vs. Malamute) are metrically separated before being committed to FAM Core.

## Background
Control run confirmed: an untrained (random init) adapter produces Condensation Ratio = 0.96 — identical to raw DINOv2 baseline. This is expected. A randomly initialized adapter is a passthrough. The triplet loss training loop must now be applied to teach the metric.

### Dataset
ImageNet G6 subset — Classes 151–200 (dog breeds), same subset used in `benchmarks/benchmark_policy_stress.py`.

### Online Hard Negative Mining
- **Anchor:** Image A, Label X
- **Positive:** Image B, Label X (same class, different sample)
- **Negative:** Image C, Label Y where `Sim(Anchor, C)` is the **highest** among all non-X candidates in the batch (hard negative)

### Training Loop
```python
adapter = MetricAdapter(input_dim=1024, output_dim=256, hidden_dim=512)
optimizer = torch.optim.Adam(adapter.parameters(), lr=1e-4)
loss_fn = TripletMarginLoss(margin=0.2)
# Train 5–10 epochs, log loss per epoch
# Save trained weights to adapter_trained.pt
```

### Evaluation
After training, re-run `benchmarks/benchmark_policy_stress.py --use-adapter` with the trained adapter loaded.

## Kill Condition
Condensation Ratio > 0.50 on the G6 (Dogs/Birds) subset after Triplet Loss training = **G11 Kill Condition Met**.

## Success Criteria
Condensation Ratio < 0.50 with trained adapter = **G11 Passed**.
